### ISSUE TITLE
Systemtray shows up only after trayer was started

### ISSUE THREAD

**Issue by Ambrevar** [ch11ng/exwm#290]
I have this in my config:
```
(require 'exwm-systemtray)
(exwm-systemtray-enable)
(setq exwm-systemtray-height 16)
```
The systemtray does not show up on startup nor when I start tray programs manually. (E.g. nm-applet, pidgin.)

It does show up, however, if I fire up `trayer` (no argument needed).
I can then close trayer and EXWM's systemtray will remain.

A wild guess would be that the systemtray gets covered by the Emacs window.

**Comment by Ambrevar** [comment]
Another wild guess is that it could be a workspace misconfiguration acording to the wiki:
>  The system tray is displayed on active workspace only.

I don't use workspaces (I have only workspace 0).

**Comment by ch11ng** [comment]
Did nm-applet really started? This application seems working for me.  You can check with `xwininfo -root -tree | grep 'EXWM system tray embedder' -A5` which would show you the tray itself and its children.

**Comment by Ambrevar** [comment]
Yes, it does get started, as confirmed by a `ps` and your command:
```
> xwininfo -root -tree | grep 'EXWM system tray embedder' -A5
                 0x1600002 "EXWM system tray embedder": ()  20x16+1900+1064  +1900+1085
                    1 child:
                    0x1a0000b "NetworkManager Applet": ("nm-applet" "Nm-applet")  16x16+2+0  +1902+1085
                       1 child:
                       0x1a0000c (has no name): ()  1x1+-1+-1  +1901+1084
              0x10000b0 (has no name): ()  1x1+-1+-1  +-1+20
```

After a fresh login, if I start nm-applet it is hidden. I can kill it and restart it, it remains hidden. Once I start `trayer` and kill it, `nm-applet` will always display properly in the systemtray.

**Comment by ch11ng** [comment]
I noticed the Emacs frame had a vertical offset of 21 pixels.  Was there something on the top of the screen? I think EXWM has a problem in calculating the workareas.

**Comment by Ambrevar** [comment]
You are correct: I launch `dzen -p -dock` before starting Emacs in my `.xinitrc`.  If I comment it out, the systemtray properly displays out of the box.

**Comment by Ambrevar** [comment]
Launching `dzen` after Emacs has started also works around the issue.  It's less convenient though.

**Comment by ch11ng** [comment]
It's fixed in 7bc4ea5a6ee1d6d2948e8ba7e065be8ac33b8a47.
